gctree/isotype.py

```diff
@@ -0,0 +1,432 @@
+            "sequence names (the original sequence ids referenced as values in\n"
+            '`idmapfile`) with format "Original SeqID, Isotype", this utility\n\n'
+            "* Adds observed isotypes to each observed node in the collapsed\n"
+            "  trees output by gctree inference. If cells with the same sequence\n"
+            "  but different isotypes are observed, then collapsed tree nodes\n"
+            "  must be ‘exploded’ into new nodes with the appropriate isotypes\n"
+            "  and abundances. Each unique sequence ID generated by gctree is\n"
+            "  prepended to its observed isotype, and a new `isotyped.idmap`\n"
+            "  mapping these new sequence IDs to original sequence IDs is \n"
+            "  written in the output directory.\n"
+            "* Resolves isotypes of unobserved ancestral genotypes in a way\n"
+            "  that minimizes isotype switching and obeys isotype switching\n"
+            "  order. If observed isotypes of an observed internal node and its\n"
+            "  children violate switching order, then the observed internal node\n"
+            "  is replaced with an unobserved node with the same sequence, and\n"
+            "  the observed internal node is placed as a child leaf. This\n"
+            "  procedure always allows switching order conflicts to be resolved,\n"
+            "  and should usually increase isotype transitions required in the\n"
+            "  resulting tree.\n"
+            "* Renders each new collapsed tree with colors and labels\n"
+            "  reflecting observed or inferred isotypes, and writes a fasta and\n"
+            "  newick file just like the gctree inference pipeline.\n"
+            "* Prints for each collapsed tree, the original branching process\n"
+            "  log likelihood, the original node count, the isotype parsimony\n"
+            "  score, and the new node count after isotype additions. The\n"
+            "  isotype parsimony score is just a count of how many isotype\n"
+            "  transitions are required along tree edges. Changes in node count\n"
+            "  after isotype additions indicate that either observed nodes had\n"
+            "  to be exploded based on observed isotypes, or isotype switching\n"
+            "  order violations required internal nodes to be expanded as leaf\n"
+            "  nodes.\n\n"
+            "This tool doesn’t make any judgements about which tree is best.\n"
+            "Tree output order is the same as in gctree inference: ranking is\n"
+            "by log likelihood before isotype additions. A determination of\n"
+            "which is the best tree is left to the user, based on likelihoods,\n"
+            "isotype parsimony score, and changes in the number of nodes after\n"
+            "isotype additions.\n"
+        ),
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    parser.add_argument(
+        "--parsimony_forest",
```

Comment:
Shouldn't this be a positional arg, rather than a named arg? Maybe a few others should be too.
```suggestion
        "parsimony_forest",
```

Reply:
This was something I wasn't sure about. Perhaps the positional arg should be a directory containing gctree outputs, and the utility can look for standard names of the things it needs, or have the user supply nonstandard names in named arguments? Or should I just change all the required things (pickled collapsed forest, log file containing parameters, idmap, and isotype map) to positional arguments to avoid confusion when names are nonstandard?

Reply:
I would just make them positional args so any filename could be used.